addons/account_analytic_default/models/sale_order_line.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    # Method overridden to set the analytic account by default on criterion match
+    @api.multi
+    def invoice_line_create(self, invoice_id, quantity):
```

Comment:
instead of override invoice_line_create method, override _prepare_invoice_line method